### DISCUSSION TITLE
10 seconds latency between live stream and HLS

### DISCUSSION THREAD

**Discussion by YalamareddyChaitanya** [bluenviron/mediamtx#529] [Questions]
Hi,

Thanks for the library. We are currently sending the live video from IOT camera to RTSP server. Then we are showing the live video in the web browser using HLS URL. We found out that there is some latency (around 10 seconds) between live stream and HLS stream on the web browser. We would like to reduce the latency as much as possible. Is there any way to reduce the latency to 1 or 2 seconds?

Thanks in advance,
Chaitanya

**Comment by aler9** [comment]
Hello, there are multiple ways

1) re-encode the stream with a lower I-frame interval (https://stackoverflow.com/questions/30979714/how-to-change-keyframe-interval). This reduces the latency to 2-3 secs.
2 For even lower latency, use WebRTC or WASM (see here #569)

[a comment by an automated account is not shown]
